### ISSUE TITLE
在进行模型推断时是否还在进行梯度的更新？

### ISSUE THREAD

**Issue by zyx1017** [RUCAIBox/RecBole#1804]
你好！感谢你们的工作！
我有个疑问，伯乐框架中模型的推断代码一般如下：以SASRec为例子

def predict(self, interaction):
        item_seq = interaction[self.ITEM_SEQ]
        item_seq_len = interaction[self.ITEM_SEQ_LEN]
        test_item = interaction[self.ITEM_ID]
        seq_output = self.forward(item_seq, item_seq_len)
        test_item_emb = self.item_embedding(test_item)
        scores = torch.mul(seq_output, test_item_emb).sum(dim=1)  # [B]
        return scores

 def full_sort_predict(self, interaction):
        item_seq = interaction[self.ITEM_SEQ]
        item_seq_len = interaction[self.ITEM_SEQ_LEN]
        seq_output = self.forward(item_seq, item_seq_len)
        test_items_emb = self.item_embedding.weight
        scores = torch.matmul(seq_output, test_items_emb.transpose(0, 1))  # [B n_items]
        return scores

而我看一些其他作者写的代码，一般在做预测时会停止梯度的更新：
def predict(self,seq, pos,seq_len):
        with torch.no_grad():
            seq_emb = self.model.forward(seq,pos)
            last_item_embeddings = [seq_emb[i,last-1,:].view(-1,self.emb_size) for i,last in enumerate(seq_len)]
            score = torch.matmul(torch.cat(last_item_embeddings,0), self.model.item_emb.transpose(0, 1))
        return score.cpu().numpy()

伯乐的实现方法是否存在什么问题呢？

**Comment by zyx1017** [comment]
@Sherry-XLL

**Comment by zyx1017** [comment]
抱歉！问题已解决
